### PR TITLE
Fix MEMFS use in ENVIRONMENT=web + closure. Fixes #8010

### LINKED ISSUE #8010: ENVIRONMENT=web, optimised builds, Closure and stdio
We're seeing an issue which I've narrowed down to Closure use on optimised builds with stdio. I'm guesing it's related to #7933 and #7803, but here's a standalone reproducible case nonetheless:
```
#include <cstdio>
int main() {
	fopen("test", "wb");
	return 0;
}
```
Build with:
```
emcc -s ENVIRONMENT=web -s STRICT -Os --closure 1 test.cpp
```
It will fail with Closure's dump followed by `4 error(s), 2 warning(s)`. Remove the `ENVIRONMENT` flag or set it to `node`, or remove the `-O` option, and it's fine. This was fine in 1.38.24 but fails after.

### PR DESCRIPTION
**Pull request by juj**
Also clean up Closure warnings about duplicate switch-case values.

### CODE REVIEW

File: src/library_syscall.js [hunk 1]
```diff
@@ -1118,7 +1118,10 @@ var SyscallsLibrary = {
         return 0;
       }
       case {{{ cDefine('F_GETLK') }}}:
-      case {{{ cDefine('F_GETLK64') }}}: {
+      /* case {{{ cDefine('F_GETLK64') }}}: Currently in musl F_GETLK64 has same value as F_GETLK, but if that changes, uncomment this */ {
+#if ASSERTIONS
+        assert({{{ cDefine('F_GETLK') }}} === {{{ cDefine('F_GETLK64') }}});
```
Comment (kripken) [inline src/library_syscall.js]: let's make this a static assert, that is, that runs at compile time, and does not depend on this code being reached at runtime. we can do it like this:
```
{{{ assert(cDefine('F_GETLK') === cDefine('F_GETLK64')), '' }}}
```
(we need to return an empty string, hence the `''`)

File: src/library_syscall.js [hunk 2]
```diff
@@ -1118,7 +1118,10 @@ var SyscallsLibrary = {
         return 0;
       }
       case {{{ cDefine('F_GETLK') }}}:
-      case {{{ cDefine('F_GETLK64') }}}: {
+      /* case {{{ cDefine('F_GETLK64') }}}: Currently in musl F_GETLK64 has same value as F_GETLK, but if that changes, uncomment this */ {
```
Comment (kripken) [inline src/library_syscall.js]: is the issue that closure would error on a duplicate key? please add that to the comment

File: src/library_syscall.js (hunk 1 above)
Comment (kripken) [reply src/library_syscall.js]: (same for the other assert later down)

File: src/library_syscall.js (hunk 2 above)
Comment (juj) [reply src/library_syscall.js]: It issues a warning, but we don't print out closure warnings, so it got lost.

File: src/library_syscall.js (hunk 1 above)
Comment (juj) [reply src/library_syscall.js]: ok

**Pull request merged**